packages/core-data/src/resolvers.js

```diff
@@ -247,39 +236,39 @@ getEntityRecords.shouldInvalidate = ( action, kind, name ) => {
 /**
  * Requests the current theme.
  */
-export function* getCurrentTheme() {
-	const activeThemes = yield apiFetch( {
+export const getCurrentTheme = () => async ( { dispatch } ) => {
+	const activeThemes = await triggerFetch( {
 		path: '/wp/v2/themes?status=active',
 	} );
-	yield receiveCurrentTheme( activeThemes[ 0 ] );
-}
+	dispatch.receiveCurrentTheme( activeThemes[ 0 ] );
+};
 
 /**
  * Requests theme supports data from the index.
  */
-export function* getThemeSupports() {
-	const activeThemes = yield apiFetch( {
+export const getThemeSupports = () => async ( { dispatch } ) => {
+	const activeThemes = await triggerFetch( {
 		path: '/wp/v2/themes?status=active',
 	} );
-	yield receiveThemeSupports( activeThemes[ 0 ].theme_supports );
-}
+	dispatch.receiveThemeSupports( activeThemes[ 0 ].theme_supports );
+};
```

Comment:
`getCurrentTheme` and `getThemeSupports` will issue a duplicate REST request, instead of sharing an existing cached response. The `getThemeSupports` resolver could be written as:
```js
const getThemeSupports = () => async ( { resolveSelect } ) => {
  const currentTheme = await resolveSelect.getCurrentTheme();
  dispatch.receiveThemeSupports( currentTheme.theme_supports );
};
```
The next step would be to ditch `receiveThemeSupports` and the associated state, and implement the `getThemeSupports` selector in terms of `getCurrentTheme`.

Can be a separate PR, doesn't prevent merging this one as-is at all.

Reply:
Good point! I think it issues a duplicate REST request already and that would be a good quality of life improvement 👍